### PR TITLE
ci: add 3 plugin test cases for e2e

### PR DESCRIPTION
**Pull request by AlinsRan**
### Type of change:

- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [x] Backport patches

#### Test cases include:
* **consumer-restriction plugin**
* **csrf plugin**
* **client-control plugin**
### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Comment by tao12345666333** [comment]
Please run `make e2e-names-check` to check e2e test case name

File: test/e2e/suite-plugins/consumer-restriction.go [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+		ac1 := CreateBasicAuthConsumer("jack1", "jack1-username", "jack1-password")
+		assert.Nil(ginkgo.GinkgoT(), s.CreateResourceFromString(ac1), "creating basicAuth ApisixConsumer")
+
+		ac2 := CreateBasicAuthConsumer("jack2", "jack2-username", "jack2-password")
+		assert.Nil(ginkgo.GinkgoT(), s.CreateResourceFromString(ac2), "creating basicAuth ApisixConsumer")
+
+		// Wait until the ApisixConsumer create event was delivered.
+		time.Sleep(6 * time.Second)
+
+		grs, err := s.ListApisixConsumers()
```
Comment (tao12345666333) [inline test/e2e/suite-plugins/consumer-restriction.go:78]: What is the meaning of `grs`?

**Review by tao12345666333 — APPROVED** [review]
LGTM

Just left a comment.

File: test/e2e/suite-plugins/consumer-restriction.go [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+//consumer-restriction plugin testing
+var _ = ginkgo.Describe("suite-plugins: consumer-restriction plugin", func() {
+	opts := &scaffold.Options{
+		Name:                  "default",
+		Kubeconfig:            scaffold.GetKubeconfig(),
+		APISIXConfigPath:      "testdata/apisix-gw-config.yaml",
+		IngressAPISIXReplicas: 1,
+		HTTPBinServicePort:    80,
+		APISIXRouteVersion:    "apisix.apache.org/v2beta3",
```
Comment (gxthrj) [inline test/e2e/suite-plugins/consumer-restriction.go:64]: Should we use `apisix.apache.org/v2` now? cc @tao12345666333

Comment (tao12345666333) [reply test/e2e/suite-plugins/consumer-restriction.go:64]: > Should we use `apisix.apache.org/v2` now? cc @tao12345666333

good catch! @AlinsRan please change this to `apisix.apache.org/v2` 
thanks!

Comment (AlinsRan) [reply test/e2e/suite-plugins/consumer-restriction.go:64]: why? How are they diff? Does this need to be done for subsequent test cases.

Comment (AlinsRan) [reply test/e2e/suite-plugins/consumer-restriction.go:64]: @tao12345666333

Comment (tao12345666333) [reply test/e2e/suite-plugins/consumer-restriction.go:64]: > why? How are they diff? Does this need to be done for subsequent test cases. 

v2 is the latest stable version. v2 is also required in subsequent test cases

Comment (AlinsRan) [reply test/e2e/suite-plugins/consumer-restriction.go:64]: > > why? How are they diff? Does this need to be done for subsequent test cases.
> 
> v2 is the latest stable version. v2 is also required in subsequent test cases

ok

Comment (tao12345666333) [reply test/e2e/suite-plugins/consumer-restriction.go:64]: In fact, we haven't released the v2 version of CRD, so there is no need to modify it here. 😅

@gxthrj @AlinsRan

File: test/e2e/suite-plugins/consumer-restriction.go [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+	"net/http"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/apache/apisix-ingress-controller/test/e2e/scaffold"
+)
+
+func CreateBasicAuthConsumer(name, username, password string) string {
```
Comment (Chever-John) [inline test/e2e/suite-plugins/consumer-restriction.go:29]: Hi, @tao12345666333 @AlinsRan 
I don't think the function here should be pulled out separately. You could put it in the scaffold folder(or file) if you wanted to do that. Then you can use it by calling the function, but that would mean that every test file should do that and the scaffold folder would become bloated.

I know **[DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)** pricinple. But maybe we do not need it here.

File: test/e2e/suite-plugins/client-control.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+ - name: rule1
+   match:
+     hosts:
+     - httpbin.org
+     paths:
+       - /anything
+   backends:
+   - serviceName: %s
+     servicePort: %d
+     weight: 10
```
Comment (tao12345666333) [inline test/e2e/suite-plugins/client-control.go:55]: This line is not needed, you can delete it

File: test/e2e/suite-plugins/client-control.go [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
```
Comment (tao12345666333) [inline test/e2e/suite-plugins/client-control.go:105]: ditto

File: test/e2e/suite-plugins/csrf.go [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+ - name: rule1
+   match:
+     hosts:
+     - httpbin.org
+     paths:
+       - /anything
+   backends:
+   - serviceName: %s
+     servicePort: %d
+     weight: 10
```
Comment (tao12345666333) [inline test/e2e/suite-plugins/csrf.go:55]: ditto

File: test/e2e/suite-plugins/csrf.go [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
```
Comment (tao12345666333) [inline test/e2e/suite-plugins/csrf.go:116]: ditto

**Review by tao12345666333 — COMMENTED** [review]
Thanks
Only one point needs to be adjusted, other contents LGTM

**Review by tao12345666333 — APPROVED** [review]
LGTM

**Review by lingsamuel — APPROVED** [review]

**Comment by lijing-21** [comment]
Hi @AlinsRan  , thank you for your contribution!

Here is the Contributor T-shirt form[1], if you're interested, kindly take a look :)

[1] https://github.com/apache/apisix/blob/master/CONTRIBUTING.md#contributor-t-shirt

**Pull request merged**